### PR TITLE
minior improve ecds generate

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

- refactor secretConfigsUpdatedOnly
- skip when only secret updated but no referenced secret

### CODE REVIEW

File: pilot/pkg/xds/ecds.go [hunk 1]
```diff
@@ -59,6 +59,14 @@ func ecdsNeedsPush(req *model.PushRequest) bool {
 	return false
 }
 
+// Check if the secret updates is relevant to Wasm image pull. If not relevant, skip pushing ECDS.
+// this occurs when user only update the Secret referenced by WasmPlugin/EnvoyFilter.
+func secretConfigsUpdatedOnly(req *model.PushRequest) bool {
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/ecds.go:64]: I would prefer also to abstract `updatedSecrets` cal into this function and rename it, as it is very confusing

Comment (zirain) [reply pilot/pkg/xds/ecds.go:64]: updated, PTAL.

File: pilot/pkg/xds/ecds.go [hunk 2]
```diff
@@ -59,65 +59,92 @@ func ecdsNeedsPush(req *model.PushRequest) bool {
...
+
+	return resources, model.DefaultXdsLogDetails, nil
+}
+
+func (e *EcdsGenerator) pullSecrets(proxy *model.Proxy, w *model.WatchedResource, req *model.PushRequest) (secrets map[string][]byte, needsPush bool) {
 	secretResources := referencedSecrets(proxy, req.Push, w.ResourceNames)
-	// Check if the secret updates is relevant to Wasm image pull. If not relevant, skip pushing ECDS.
-	if !model.ConfigsHaveKind(req.ConfigsUpdated, kind.WasmPlugin) && !model.ConfigsHaveKind(req.ConfigsUpdated, kind.EnvoyFilter) &&
-		model.ConfigsHaveKind(req.ConfigsUpdated, kind.Secret) {
+	if secretConfigsUpdatedOnly(req) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/ecds.go:99]: IMO, this is an optimization to skip push ecds only when secret update, but the secret is not relevant. Add more comments

File: pilot/pkg/xds/ecds.go [hunk 3]
```diff
@@ -59,65 +59,92 @@ func ecdsNeedsPush(req *model.PushRequest) bool {
...
+		resources = append(resources, &discovery.Resource{
+			Name:     c.Name,
+			Resource: protoconv.MessageToAny(c),
+		})
+	}
+
+	return resources, model.DefaultXdsLogDetails, nil
+}
+
+func (e *EcdsGenerator) pullSecrets(proxy *model.Proxy, w *model.WatchedResource, req *model.PushRequest) (secrets map[string][]byte, needsPush bool) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/ecds.go:97]: The function name is confusing

File: pilot/pkg/xds/ecds.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/xds/ecds.go:99]: Maybe better to move the if clause out

**Pull request closed without merging**